ext/DataInterpolationsNDSymbolicsExt.jl

```diff
@@ -0,0 +1,100 @@
+module DataInterpolationsNDSymbolicsExt
+
+using DataInterpolationsND: NDInterpolation
+using Symbolics
+using Symbolics: Num, unwrap, SymbolicUtils
+
+# Register just one symbolic function - the promote_symtype is handled by the macro
+@register_symbolic (interp::NDInterpolation)(t::Real)
+
+Base.nameof(interp::NDInterpolation) = :NDInterpolation
+
+# Add method to handle multiple arguments symbolically  
+function (interp::NDInterpolation)(args::Vararg{Num})
```

Comment:
Easiest to only support the all Num case, can `Union{Number,Num}` but then all other dispatches need to `::Number`.

Reply:
It also needs to define `::Vararg{BasicSymbolic{<:Real}}`